integration-ambient tests fail: istio-beacon-k8s stuck "Validating waypoint readiness" and kserve-controller blocked

### Bug Description

After fixing the root `tox.ini` factor routing so that `kserve-controller-integration-ambient`
actually runs `tests/integration/test_charm_ambient.py` (previously it silently fell through to the
`integration` env and ran `test_charm.py`), the ambient integration suite now executes but fails in
`test_build_and_deploy`.

Two problems are observed:

1. `istio-beacon-k8s/0` first reports `hook failed: "config-changed"`, then becomes stuck
   indefinitely in `maintenance: Validating waypoint readiness`, causing
   `model.wait_for_idle(...)` to raise `asyncio.TimeoutError` after ~17 minutes.
2. `kserve-controller/0` remains `blocked` with:
   `Standard mode detected, but no relation to gateway-metadata or ingress-gateway`.

Final result: `1 failed, 19 xfailed` (all remaining tests xfail because deploy never succeeds).

Note on the original tox bug (root cause of it never running before): in the root `tox.ini`,
`integration` is a sub-factor of `integration-ambient`, so the line
`integration: TYPE = integration` matched and set `TYPE=integration`. With no
`integration-ambient: TYPE = integration-ambient` mapping, the ambient env always ran the standard
`integration` env. Adding that mapping is what exposed the failure below.


### To Reproduce

1. Ensure the root `tox.ini` maps the ambient factor:
   `integration-ambient: TYPE = integration-ambient`
2. From the repo root run:
   `tox -e kserve-controller-integration-ambient -- --model kubeflow --charm-path=<path-to>/kserve-controller_ubuntu@24.04-amd64.charm`
3. Observe `istio-beacon-k8s/0` stuck in `Validating waypoint readiness` and the test timing out
   in `test_build_and_deploy`, with `kserve-controller` blocked on missing gateway relation.


### Environment

- Platform: GitHub Actions runner (Ubuntu), concierge-managed k8s.
- Juju: model controller `concierge-k8s`, k8s cloud, Juju version `3.6.23`.
- Charms deployed:
  - `kserve-controller`: local build (`ubuntu@24.04-amd64`)
  - `istio-k8s` rev 45 (`2/stable`)
  - `istio-ingress-k8s` rev 61 (`2/stable`)
  - `istio-beacon-k8s` rev 63 (`2/stable`)
- Python 3.12.3, pytest 8.4.2, pytest-operator 0.42.1, pytest-asyncio 0.21.2.
- Run reference: [GitHub Actions run 27533326648 / job 81378075241.](https://github.com/canonical/kserve-operators/actions/runs/27533326648/job/81378075241)


### Relevant Log Output

```shell
istio-beacon-k8s/0 [idle] error: hook failed: "config-changed"
istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness   (repeats until timeout)

kserve-controller/0*  blocked  idle  Standard mode detected, but no relation to gateway-metadata or ingress-gateway

unit-istio-beacon-k8s-0: ERROR unit.istio-beacon-k8s/0.juju-log Uncaught exception while in charm code:
  File ".../src/charm.py", line 567, in mesh_labels_for_service_mesh_relation
    if self.config["model-on-mesh"]:
  ...
ops.model.ModelError: ERROR read tcp 10.1.0.202:46314->10.152.183.213:17070: read: connection reset by peer
juju.worker.uniter.operation hook "config-changed" (via hook dispatching script: dispatch) failed: exit status 1

tests/integration/test_charm_ambient.py::test_build_and_deploy
  File ".../charmed_kubeflow_chisme/testing/ambient_integration.py", line 89, in deploy_and_integrate_service_mesh_charms
    await model.wait_for_idle(...)
  File ".../juju/model/__init__.py", line 3250, in wait_for_idle
    raise asyncio.TimeoutError(
TimeoutError: Timed out waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness

============ 1 failed, 19 xfailed, 3 warnings in 1037.93s (0:17:17) ============
```

### Additional Context

_No response_

The observed failure still exhibits sporadically, as shown [here](https://github.com/canonical/kserve-operators/actions/runs/27555719164/job/81611693643):

<details>
<summary>logs</summary>

```log
juju add-model kubeflow --config automatically-retry-hooks=false
  tox -e kserve-controller-integration-ambient -- --model kubeflow --charm-path=/home/runner/work/kserve-operators/kserve-operators/charms/kserve-controller/kserve-controller_ubuntu@24.04-amd64.charm
  shell: /usr/bin/bash -e {0}
Added 'kubeflow' model with credential 'k8s' for user 'admin'
kserve-controller-integration-ambient: install_deps> python -I -m pip install 'poetry>=2.1.3'
kserve-controller-integration-ambient: freeze> python -m pip freeze --all
kserve-controller-integration-ambient: anyio==4.14.0,backports.zstd==1.6.0,build==1.5.0,CacheControl==0.14.4,certifi==2026.5.20,cffi==2.0.0,charset-normalizer==3.4.7,cleo==2.1.0,crashtest==0.4.1,cryptography==49.0.0,distlib==0.4.3,dulwich==1.2.6,fastjsonschema==2.21.2,filelock==3.29.4,findpython==0.8.0,h11==0.16.0,httpcore==1.0.9,httpx==0.28.1,idna==3.18,installer==1.0.1,jaraco.classes==3.4.0,jaraco.context==6.1.2,jaraco.functools==4.5.0,jeepney==0.9.0,keyring==25.7.0,more-itertools==11.1.0,msgpack==1.2.0,packaging==26.2,pbs-installer==2026.6.10,pip==26.1.2,pkginfo==1.12.1.2,platformdirs==4.10.0,poetry==2.4.1,poetry-core==2.4.0,pycparser==3.0,pyproject_hooks==1.2.0,python-discovery==1.4.2,RapidFuzz==3.14.5,requests==2.34.2,requests-toolbelt==1.0.0,SecretStorage==3.5.0,shellingham==1.5.4,tomlkit==0.15.0,trove-classifiers==2026.6.1.19,typing_extensions==4.15.0,urllib3==2.7.0,virtualenv==21.5.0
kserve-controller-integration-ambient: commands[0]> tox -c charms/kserve-controller -e integration-ambient -- --model kubeflow --charm-path=/home/runner/work/kserve-operators/kserve-operators/charms/kserve-controller/kserve-controller_ubuntu@24.04-amd64.charm
integration-ambient: install_deps /home/runner/work/kserve-operators/kserve-operators/charms/kserve-controller> python -I -m pip install 'poetry>=2.1.3'
integration-ambient: commands_pre[0] /home/runner/work/kserve-operators/kserve-operators/charms/kserve-controller> poetry install --only integration
Installing dependencies from lock file

Package operations: 80 installs, 2 updates, 0 removals

  - Installing aiohappyeyeballs (2.6.2)
  - Installing aiohttp (3.14.1)
  - Installing aiosignal (1.4.0)
  - Installing annotated-types (0.7.0)
  - Downgrading anyio (4.14.0 -> 4.13.0)
  - Installing asttokens (3.0.1)
  - Installing attrs (26.1.0)
  - Installing backports-datetime-fromisoformat (2.0.3)
  - Installing bcrypt (5.0.0)
  - Installing charmed-kubeflow-chisme (0.4.25)
  - Installing charmed-service-mesh-helpers (0.3.0)
  - Installing decorator (5.3.1)
  - Installing deepdiff (6.2.1)
  - Installing executing (2.2.1)
  - Installing frozenlist (1.8.0)
  - Installing google-auth (2.54.0)
  - Downgrading httpx (0.28.1 -> 0.27.2)
  - Installing hvac (2.4.0)
  - Installing importlib-metadata (9.0.0)
  - Installing iniconfig (2.3.0)
  - Installing invoke (3.0.3)
  - Installing ipdb (0.13.13)
  - Installing ipython (9.14.1)
  - Installing ipython-pygments-lexers (1.1.1)
  - Installing jedi (0.20.0)
  - Installing jinja2 (3.1.6)
  - Installing jsonschema (4.17.3)
  - Installing juju (3.6.1.3)
  - Installing kubernetes (30.1.0)
  - Installing lightkube (0.17.1)
  - Installing lightkube-extensions (0.3.0)
  - Installing lightkube-models (1.35.0.8)
  - Installing macaroonbakery (1.3.4)
  - Installing markupsafe (3.0.3)
  - Installing matplotlib-inline (0.2.2)
  - Installing multidict (6.7.1)
  - Installing mypy-extensions (1.1.0)
  - Installing oauthlib (3.3.1)
  - Installing opentelemetry-api (1.42.1)
  - Installing ops (2.23.2)
  - Installing ordered-set (4.1.0)
  - Installing paramiko (5.0.0)
  - Installing parso (0.8.7)
  - Installing pexpect (4.9.0)
  - Installing pluggy (1.6.0)
  - Installing prompt-toolkit (3.0.52)
  - Installing propcache (0.5.2)
  - Installing protobuf (7.35.1)
  - Installing psutil (7.2.2)
  - Installing ptyprocess (0.7.0)
  - Installing pure-eval (0.2.3)
  - Installing pyasn1 (0.6.3)
  - Installing pyasn1-modules (0.4.2)
  - Installing pydantic (2.13.4)
  - Installing pydantic-core (2.46.4)
  - Installing pygments (2.20.0)
  - Installing pymacaroons (0.13.0)
  - Installing pynacl (1.6.2)
  - Installing pyrfc3339 (1.1)
  - Installing pyrsistent (0.20.0)
  - Installing pytest (8.4.2)
  - Installing pytest-asyncio (0.21.2)
  - Installing pytest-operator (0.42.1)
  - Installing python-dateutil (2.9.0.post0)
  - Installing pytz (2026.2)
  - Installing pyyaml (6.0.1)
  - Installing requests-oauthlib (2.0.0)
  - Installing ruamel-yaml (0.19.1)
  - Installing serialized-data-interface (0.7.0)
  - Installing six (1.17.0)
  - Installing sniffio (1.3.1)
  - Installing stack-data (0.6.3)
  - Installing tenacity (9.1.4)
  - Installing toposort (1.10)
  - Installing traitlets (5.15.1)
  - Installing typing-inspect (0.9.0)
  - Installing typing-inspection (0.4.2)
  - Installing wcwidth (0.8.1)
  - Installing websocket-client (1.9.0)
  - Installing websockets (16.0)
  - Installing yarl (1.24.2)
  - Installing zipp (4.1.0)
integration-ambient: commands[0] /home/runner/work/kserve-operators/kserve-operators/charms/kserve-controller> pytest -v --tb native --ignore=/home/runner/work/kserve-operators/kserve-operators/charms/kserve-controller/tests/unit --log-cli-level=INFO -s --asyncio-mode=auto /home/runner/work/kserve-operators/kserve-operators/charms/kserve-controller/tests/integration/test_charm_ambient.py --model kubeflow --charm-path=/home/runner/work/kserve-operators/kserve-operators/charms/kserve-controller/kserve-controller_ubuntu@24.04-amd64.charm
============================= test session starts ==============================
platform linux -- Python 3.12.3, pytest-8.4.2, pluggy-1.6.0 -- /home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/bin/python
cachedir: /home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/.pytest_cache
rootdir: /home/runner/work/kserve-operators/kserve-operators/charms/kserve-controller
configfile: pyproject.toml
plugins: asyncio-0.21.2, anyio-4.13.0, operator-0.42.1
asyncio: mode=Mode.AUTO
collecting ... collected 20 items

tests/integration/test_charm_ambient.py::test_build_and_deploy 
-------------------------------- live log setup --------------------------------
WARNING  juju.client.connection:connection.py:858 unexpected facade SSHServer received from the controller
INFO     pytest_operator.plugin:plugin.py:795 Connecting to existing model concierge-k8s:kubeflow on unspecified cloud
WARNING  juju.client.connection:connection.py:858 unexpected facade SSHServer received from the controller
WARNING  juju.client.connection:connection.py:858 unexpected facade SSHSession received from the controller
WARNING  juju.client.connection:connection.py:858 unexpected facade SSHTunneler received from the controller
WARNING  juju.client.connection:connection.py:858 unexpected facade SecretsRevoker received from the controller
WARNING  juju.client.connection:connection.py:858 unexpected facade SSHServer received from the controller
-------------------------------- live log call ---------------------------------
INFO     juju.model:__init__.py:2301 Deploying local:kserve-controller-0
INFO     juju.model:__init__.py:2301 Deploying ch:amd64/noble/istio-k8s-45
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-k8s/0 [allocating] waiting: installing agent
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-k8s/0 [idle] active: 
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-k8s/0 [idle] active: 
INFO     juju.model:__init__.py:2301 Deploying ch:amd64/noble/istio-ingress-k8s-61
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-ingress-k8s/0 [allocating] waiting: installing agent
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-ingress-k8s/0 [executing] active: Serving at 10.64.140.43
INFO     juju.model:__init__.py:2301 Deploying ch:amd64/jammy/istio-beacon-k8s-63
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [allocating] waiting: installing agent
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] error: hook failed: "config-changed"
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
FAILED
tests/integration/test_charm_ambient.py::test_inference_service[./tests/integration/crs/predictionserver-sklearn.yaml] XFAIL
tests/integration/test_charm_ambient.py::test_inference_service[./tests/integration/crs/sklearn-iris.yaml] XFAIL
tests/integration/test_charm_ambient.py::test_inference_service[./tests/integration/crs/lgbserver.yaml] XFAIL
tests/integration/test_charm_ambient.py::test_inference_service[./tests/integration/crs/pmml-server.yaml] XFAIL
tests/integration/test_charm_ambient.py::test_inference_service[./tests/integration/crs/paddleserver-resnet.yaml] XFAIL
tests/integration/test_charm_ambient.py::test_inference_service[./tests/integration/crs/xgbserver.yaml] XFAIL
tests/integration/test_charm_ambient.py::test_inference_service[./tests/integration/crs/tensorflow-serving.yaml] XFAIL
tests/integration/test_charm_ambient.py::test_logging XFAIL (aborted)
tests/integration/test_charm_ambient.py::test_metrics_enpoint XFAIL
tests/integration/test_charm_ambient.py::test_alert_rules XFAIL (abo...)
tests/integration/test_charm_ambient.py::test_configmap_created XFAIL
tests/integration/test_charm_ambient.py::test_configmap_changes_with_config XFAIL
tests/integration/test_charm_ambient.py::test_relate_to_object_store XFAIL
tests/integration/test_charm_ambient.py::test_deploy_resource_dispatcher XFAIL
tests/integration/test_charm_ambient.py::test_new_user_namespace_has_manifests XFAIL
tests/integration/test_charm_ambient.py::test_inference_service_proxy_envs_configuration XFAIL
tests/integration/test_charm_ambient.py::test_blocked_on_invalid_config XFAIL
tests/integration/test_charm_ambient.py::test_container_security_context[kserve-controller] XFAIL
tests/integration/test_charm_ambient.py::test_container_security_context[charm] XFAIL
------------------------------ live log teardown -------------------------------
INFO     pytest_operator.plugin:plugin.py:937 Model status:

Model     Controller     Cloud/Region  Version  SLA          Timestamp
kubeflow  concierge-k8s  k8s           3.6.23   unsupported  08:39:46Z

App                Version  Status       Scale  Charm              Channel   Rev  Address         Exposed  Message
istio-beacon-k8s            maintenance      1  istio-beacon-k8s   2/stable   63  10.152.183.243  no       Validating waypoint readiness
istio-ingress-k8s           active           1  istio-ingress-k8s  2/stable   61  10.152.183.171  no       Serving at 10.64.140.43
istio-k8s          1.26.1   active           1  istio-k8s          2/stable   45  10.152.183.226  no       
kserve-controller           blocked          1  kserve-controller              0  10.152.183.205  no       Standard mode detected, but no relation to gateway-metadata or ingress-gateway

Unit                  Workload     Agent  Address     Ports  Message
istio-beacon-k8s/0*   maintenance  idle   10.1.0.245         Validating waypoint readiness
istio-ingress-k8s/0*  active       idle   10.1.0.12          Serving at 10.64.140.43
istio-k8s/0*          active       idle   10.1.0.194         
kserve-controller/0*  blocked      idle   10.1.0.173         Standard mode detected, but no relation to gateway-metadata or ingress-gateway


INFO     pytest_operator.plugin:plugin.py:943 Juju error logs:

unit-kserve-controller-0: 08:22:46 ERROR juju.worker.proxyupdater error writing apt proxy config file: open /etc/apt/apt.conf.d/95-juju-proxy-settings: permission denied
unit-kserve-controller-0: 08:22:52 ERROR unit.kserve-controller/0.juju-log Failed to handle <InstallEvent via KServeControllerCharm/on/install[1]> with error: Standard mode detected, but no relation to gateway-metadata or ingress-gateway
unit-kserve-controller-0: 08:22:53 ERROR unit.kserve-controller/0.juju-log Failed to handle <LeaderElectedEvent via KServeControllerCharm/on/leader_elected[6]> with error: Standard mode detected, but no relation to gateway-metadata or ingress-gateway
unit-kserve-controller-0: 08:22:54 ERROR unit.kserve-controller/0.juju-log Failed to handle <PebbleReadyEvent on <ops.model.Container object at 0x7f9cc48a26c0> via KServeControllerCharm/on/kserve_controller_pebble_ready[11]> with error: Standard mode detected, but no relation to gateway-metadata or ingress-gateway
unit-kserve-controller-0: 08:22:55 ERROR unit.kserve-controller/0.juju-log Failed to handle <ConfigChangedEvent via KServeControllerCharm/on/config_changed[16]> with error: Standard mode detected, but no relation to gateway-metadata or ingress-gateway
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.worker.caasunitterminationworker error while terminating unit: codec.ReadHeader error: receiving message: read tcp 10.1.0.242:32774->10.152.183.179:17070: read: connection reset by peer
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: codec.ReadHeader error: receiving message: read tcp 10.1.0.242:32774->10.152.183.179:17070: read: connection reset by peer
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-beacon-k8s-0: 08:25:06 ERROR juju.worker.uniter.operation after executing operation "run config-changed hook" for istio-beacon-k8s/0 for istio-beacon-k8s/0: writing state: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: codec.ReadHeader error: receiving message: read tcp 10.1.0.194:46150->10.152.183.179:17070: read: connection reset by peer
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-k8s-0: 08:25:13 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-kserve-controller-0: 08:25:16 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: codec.ReadHeader error: receiving message: read tcp 10.1.0.173:34280->10.152.183.179:17070: read: connection reset by peer
unit-kserve-controller-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
unit-istio-ingress-k8s-0: 08:25:16 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: codec.ReadHeader error: receiving message: read tcp 10.1.0.12:55114->10.152.183.179:17070: read: connection reset by peer
model-5d0731d1-151d-4647-8061-5fea057d1ab4: 08:25:17 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
model-5d0731d1-151d-4647-8061-5fea057d1ab4: 08:25:17 ERROR juju.api.watcher error trying to stop watcher: connection is shutdown before send: connection is shut down
model-5d0731d1-151d-4647-8061-5fea057d1ab4: 08:25:17 ERROR juju.worker.dependency "api-caller" manifold worker returned unexpected error: codec.ReadHeader error: receiving message: read tcp 10.1.0.11:44842->10.152.183.179:17070: read: connection reset by peer
unit-kserve-controller-0: 08:25:19 ERROR juju.worker.proxyupdater error writing apt proxy config file: open /etc/apt/apt.conf.d/95-juju-proxy-settings: permission denied

INFO     pytest_operator.plugin:plugin.py:1025 Forgetting model main...


=================================== FAILURES ===================================
____________________________ test_build_and_deploy _____________________________
Traceback (most recent call last):
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/_pytest/runner.py", line 344, in from_call
    result: TResult | None = func()
                             ^^^^^^
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/_pytest/runner.py", line 246, in <lambda>
    lambda: runtest_hook(item=item, **kwds), when=when, reraise=reraise
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/pluggy/_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/pluggy/_callers.py", line 167, in _multicall
    raise exception
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/pluggy/_callers.py", line 139, in _multicall
    teardown.throw(exception)
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/_pytest/logging.py", line 850, in pytest_runtest_call
    yield
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/pluggy/_callers.py", line 139, in _multicall
    teardown.throw(exception)
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/_pytest/capture.py", line 900, in pytest_runtest_call
    return (yield)
            ^^^^^
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/pluggy/_callers.py", line 139, in _multicall
    teardown.throw(exception)
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/_pytest/skipping.py", line 263, in pytest_runtest_call
    return (yield)
            ^^^^^
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/pluggy/_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/_pytest/runner.py", line 178, in pytest_runtest_call
    item.runtest()
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/_pytest/python.py", line 1671, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/pluggy/_hooks.py", line 512, in __call__
    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/pluggy/_manager.py", line 120, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/pluggy/_callers.py", line 167, in _multicall
    raise exception
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/pluggy/_callers.py", line 139, in _multicall
    teardown.throw(exception)
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/pluggy/_callers.py", line 53, in run_old_style_hookwrapper
    return result.get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/pluggy/_result.py", line 103, in get_result
    raise exc.with_traceback(tb)
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/pluggy/_callers.py", line 38, in run_old_style_hookwrapper
    res = yield
          ^^^^^
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/pluggy/_callers.py", line 121, in _multicall
    res = hook_impl.function(*args)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/_pytest/python.py", line 157, in pytest_pyfunc_call
    result = testfunction(**testargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/pytest_asyncio/plugin.py", line 529, in inner
    _loop.run_until_complete(task)
  File "/usr/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/runner/work/kserve-operators/kserve-operators/charms/kserve-controller/tests/integration/test_charm_ambient.py", line 114, in test_build_and_deploy
    await deploy_and_integrate_service_mesh_charms(
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/charmed_kubeflow_chisme/testing/ambient_integration.py", line 89, in deploy_and_integrate_service_mesh_charms
    await model.wait_for_idle(
  File "/home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/juju/model/__init__.py", line 3250, in wait_for_idle
    raise asyncio.TimeoutError(
TimeoutError: Timed out waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
------------------------------ Captured log setup ------------------------------
WARNING  juju.client.connection:connection.py:858 unexpected facade SSHServer received from the controller
INFO     pytest_operator.plugin:plugin.py:795 Connecting to existing model concierge-k8s:kubeflow on unspecified cloud
WARNING  juju.client.connection:connection.py:858 unexpected facade SSHServer received from the controller
WARNING  juju.client.connection:connection.py:858 unexpected facade SSHSession received from the controller
WARNING  juju.client.connection:connection.py:858 unexpected facade SSHTunneler received from the controller
WARNING  juju.client.connection:connection.py:858 unexpected facade SecretsRevoker received from the controller
WARNING  juju.client.connection:connection.py:858 unexpected facade SSHServer received from the controller
------------------------------ Captured log call -------------------------------
INFO     juju.model:__init__.py:2301 Deploying local:kserve-controller-0
INFO     juju.model:__init__.py:2301 Deploying ch:amd64/noble/istio-k8s-45
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-k8s/0 [allocating] waiting: installing agent
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-k8s/0 [idle] active: 
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-k8s/0 [idle] active: 
INFO     juju.model:__init__.py:2301 Deploying ch:amd64/noble/istio-ingress-k8s-61
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-ingress-k8s/0 [allocating] waiting: installing agent
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-ingress-k8s/0 [executing] active: Serving at 10.64.140.43
INFO     juju.model:__init__.py:2301 Deploying ch:amd64/jammy/istio-beacon-k8s-63
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [allocating] waiting: installing agent
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] error: hook failed: "config-changed"
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
INFO     juju.model:__init__.py:3254 Waiting for model:
  istio-beacon-k8s/0 [idle] maintenance: Validating waypoint readiness
=============================== warnings summary ===============================
tests/integration/test_charm_ambient.py::test_build_and_deploy
  /home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/charmed_kubeflow_chisme/testing/ambient_integration.py:60: DeprecationWarning: wait_for_active is deprecated; use status
    await model.wait_for_idle(

tests/integration/test_charm_ambient.py::test_build_and_deploy
  /home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/charmed_kubeflow_chisme/testing/ambient_integration.py:74: DeprecationWarning: wait_for_active is deprecated; use status
    await model.wait_for_idle(

tests/integration/test_charm_ambient.py::test_build_and_deploy
  /home/runner/work/kserve-operators/kserve-operators/.tox/integration-ambient/lib/python3.12/site-packages/charmed_kubeflow_chisme/testing/ambient_integration.py:89: DeprecationWarning: wait_for_active is deprecated; use status
    await model.wait_for_idle(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/integration/test_charm_ambient.py::test_build_and_deploy - TimeoutError: Timed out waiting for model:
============ 1 failed, 19 xfailed, 3 warnings in 1037.46s (0:17:17) ============
integration-ambient: exit 1 (1041.22 seconds) /home/runner/work/kserve-operators/kserve-operators/charms/kserve-controller> pytest -v --tb native --ignore=/home/runner/work/kserve-operators/kserve-operators/charms/kserve-controller/tests/unit --log-cli-level=INFO -s --asyncio-mode=auto /home/runner/work/kserve-operators/kserve-operators/charms/kserve-controller/tests/integration/test_charm_ambient.py --model kubeflow --charm-path=/home/runner/work/kserve-operators/kserve-operators/charms/kserve-controller/kserve-controller_ubuntu@24.04-amd64.charm pid=25886
  integration-ambient: FAIL code 1 (1054.46=setup[7.14]+cmd[6.10,1041.22] seconds)
  evaluation failed :( (1054.48 seconds)
kserve-controller-integration-ambient: exit 1 (1054.74 seconds) /home/runner/work/kserve-operators/kserve-operators> tox -c charms/kserve-controller -e integration-ambient -- --model kubeflow --charm-path=/home/runner/work/kserve-operators/kserve-operators/charms/kserve-controller/kserve-controller_ubuntu@24.04-amd64.charm pid=25324
  kserve-controller-integration-ambient: FAIL code 1 (1064.38=setup[9.64]+cmd[1054.74] seconds)
  evaluation failed :( (1064.41 seconds)
```

</details>

--- a/charms/kserve-controller/tests/integration/test_charm_ambient.py
+++ b/charms/kserve-controller/tests/integration/test_charm_ambient.py
@@ -7,6 +7,7 @@ import base64
 import json
 import logging
 from pathlib import Path
+from pyexpat import model
 
 import lightkube
 import lightkube.codecs
@@ -57,6 +58,14 @@ from tests.integration.utils import (
     populate_template,
 )
 
+# ambient-mode Istio:
+ISTIO_K8S_APP = "istio-k8s"
+ISTIO_INGRESS_K8S_APP = "istio-ingress-k8s"
+ISTIO_BEACON_K8S_APP = "istio-beacon-k8s"
+ISTIO_INGRESS_ROUTE_ENDPOINT = "istio-ingress-route"
+ISTIO_INGRESS_GATEWAY_ENDPOINT = "gateway-metadata"
+SERVICE_MESH_ENDPOINT = "service-mesh"
+
 # tenacity
 RETRY_FOR_THREE_MINUTES = tenacity.Retrying(
     stop=tenacity.stop_after_delay(60 * 3),
@@ -92,6 +101,32 @@ async def test_build_and_deploy(ops_test: OpsTest, request):
 
     Assert on the unit status before any relations/configurations take place.
     """
+    # deploy Istio's ecosystem:
+    istio_channel = "2/stable"
+    await ops_test.model.deploy(
+        ISTIO_K8S_APP,
+        channel=istio_channel,
+        config={"platform": ""},
+        trust=True,
+    )
+    await ops_test.model.deploy(
+        ISTIO_INGRESS_K8S_APP,
+        channel=istio_channel,
+        trust=True,
+    )
+    await ops_test.model.deploy(
+        ISTIO_BEACON_K8S_APP,
+        channel=istio_channel,
+        trust=True,
+        config={"model-on-mesh": False},
+    )
+    await ops_test.model.wait_for_idle(
+        raise_on_blocked=False,
+        raise_on_error=False,
+        wait_for_active=True,
+        timeout=900,
+    )
+
     # build and deploy charm from local source folder
     entity_url = (
         await ops_test.build_charm(".")
@@ -111,13 +146,13 @@ async def test_build_and_deploy(ops_test: OpsTest, request):
         trust=True,
     )
 
-    await deploy_and_integrate_service_mesh_charms(
-        APP_NAME,
-        model=ops_test.model,
-        channel="dev/edge",  # TODO: restore it, just a temporary channel to make the CI non-flaky
-        relate_to_beacon=True,
-        relate_to_ingress_route_endpoint=False,
-        relate_to_ingress_gateway_endpoint=True,
+    await ops_test.model.integrate(
+            f"{ISTIO_INGRESS_K8S_APP}:{ISTIO_INGRESS_ROUTE_ENDPOINT}",
+            f"{APP_NAME}:{ISTIO_INGRESS_ROUTE_ENDPOINT}",
+        )
+    await ops_test.model.integrate(
+        f"{ISTIO_BEACON_K8S_APP}:{SERVICE_MESH_ENDPOINT}",
+        f"{APP_NAME}:{SERVICE_MESH_ENDPOINT}",
     )
 
     # issuing dummy update_status just to trigger an event

--- a/charms/kserve-controller/tests/integration/test_charm_ambient.py
+++ b/charms/kserve-controller/tests/integration/test_charm_ambient.py
@@ -114,6 +114,7 @@ async def test_build_and_deploy(ops_test: OpsTest, request):
     await deploy_and_integrate_service_mesh_charms(
         APP_NAME,
         model=ops_test.model,
+        channel="dev/edge",  # TODO: restore it, just a temporary channel to make the CI non-flaky
         relate_to_beacon=True,
         relate_to_ingress_route_endpoint=False,
         relate_to_ingress_gateway_endpoint=True,

--- a/charms/kserve-controller/tests/integration/test_charm_ambient.py
+++ b/charms/kserve-controller/tests/integration/test_charm_ambient.py
@@ -92,6 +92,15 @@ async def test_build_and_deploy(ops_test: OpsTest, request):
 
     Assert on the unit status before any relations/configurations take place.
     """
+    await deploy_and_integrate_service_mesh_charms(
+        APP_NAME,
+        model=ops_test.model,
+        channel="dev/edge",  # TODO: restore it, just a temporary channel to make the CI non-flaky
+        relate_to_beacon=True,
+        relate_to_ingress_route_endpoint=False,
+        relate_to_ingress_gateway_endpoint=True,
+    )
+
     # build and deploy charm from local source folder
     entity_url = (
         await ops_test.build_charm(".")
@@ -109,15 +118,6 @@ async def test_build_and_deploy(ops_test: OpsTest, request):
         config={"deployment-mode": "standard"},
         application_name=APP_NAME,
         trust=True,
-    )
-
-    await deploy_and_integrate_service_mesh_charms(
-        APP_NAME,
-        model=ops_test.model,
-        channel="dev/edge",  # TODO: restore it, just a temporary channel to make the CI non-flaky
-        relate_to_beacon=True,
-        relate_to_ingress_route_endpoint=False,
-        relate_to_ingress_gateway_endpoint=True,
     )
 
     # issuing dummy update_status just to trigger an event

--- a/charms/kserve-controller/tests/integration/test_charm_ambient.py
+++ b/charms/kserve-controller/tests/integration/test_charm_ambient.py
@@ -92,15 +92,6 @@ async def test_build_and_deploy(ops_test: OpsTest, request):
 
     Assert on the unit status before any relations/configurations take place.
     """
-    await deploy_and_integrate_service_mesh_charms(
-        APP_NAME,
-        model=ops_test.model,
-        channel="dev/edge",  # TODO: restore it, just a temporary channel to make the CI non-flaky
-        relate_to_beacon=True,
-        relate_to_ingress_route_endpoint=False,
-        relate_to_ingress_gateway_endpoint=True,
-    )
-
     # build and deploy charm from local source folder
     entity_url = (
         await ops_test.build_charm(".")
@@ -118,6 +109,15 @@ async def test_build_and_deploy(ops_test: OpsTest, request):
         config={"deployment-mode": "standard"},
         application_name=APP_NAME,
         trust=True,
+    )
+
+    await deploy_and_integrate_service_mesh_charms(
+        APP_NAME,
+        model=ops_test.model,
+        channel="dev/edge",  # TODO: restore it, just a temporary channel to make the CI non-flaky
+        relate_to_beacon=True,
+        relate_to_ingress_route_endpoint=False,
+        relate_to_ingress_gateway_endpoint=True,
     )
 
     # issuing dummy update_status just to trigger an event

--- a/charms/kserve-controller/tests/integration/test_charm_ambient.py
+++ b/charms/kserve-controller/tests/integration/test_charm_ambient.py
@@ -7,7 +7,6 @@ import base64
 import json
 import logging
 from pathlib import Path
-from pyexpat import model
 
 import lightkube
 import lightkube.codecs

--- a/charms/kserve-controller/tests/integration/test_charm_ambient.py
+++ b/charms/kserve-controller/tests/integration/test_charm_ambient.py
@@ -58,7 +58,9 @@ from tests.integration.utils import (
 
 # ambient-mode Istio:
 ISTIO_K8S_APP = "istio-k8s"
+ISTIO_INGRESS_K8S_APP = "istio-ingress-k8s"
 ISTIO_BEACON_K8S_APP = "istio-beacon-k8s"
+ISTIO_INGRESS_GATEWAY_ENDPOINT = "gateway-metadata"
 SERVICE_MESH_ENDPOINT = "service-mesh"
 
 # tenacity
@@ -105,6 +107,11 @@ async def test_build_and_deploy(ops_test: OpsTest, request):
         trust=True,
     )
     await ops_test.model.deploy(
+        ISTIO_INGRESS_K8S_APP,
+        channel=istio_channel,
+        trust=True,
+    )
+    await ops_test.model.deploy(
         ISTIO_BEACON_K8S_APP,
         channel=istio_channel,
         trust=True,
@@ -136,6 +143,10 @@ async def test_build_and_deploy(ops_test: OpsTest, request):
         trust=True,
     )
 
+    await ops_test.model.integrate(
+        f"{ISTIO_INGRESS_K8S_APP}:{ISTIO_INGRESS_GATEWAY_ENDPOINT}",
+        f"{APP_NAME}:{ISTIO_INGRESS_GATEWAY_ENDPOINT}",
+    )
     await ops_test.model.integrate(
         f"{ISTIO_BEACON_K8S_APP}:{SERVICE_MESH_ENDPOINT}",
         f"{APP_NAME}:{SERVICE_MESH_ENDPOINT}",

--- a/charms/kserve-controller/tests/integration/test_charm_ambient.py
+++ b/charms/kserve-controller/tests/integration/test_charm_ambient.py
@@ -98,6 +98,11 @@ async def test_build_and_deploy(ops_test: OpsTest, request):
 
     Assert on the unit status before any relations/configurations take place.
     """
+    # NOTE: Chisme utilities for deploying Istio's ecosystem and integrating it with KServe are
+    # not employed because, for unclear reasons, deploying Istio after the tested application, as
+    # done in Chisme, gives sporadic, flaky problems with Beacon often stuck in maintenance status
+    # saying: "Validating waypoint readiness"
+
     # deploy Istio's ecosystem:
     istio_channel = "2/stable"
     await ops_test.model.deploy(

--- a/charms/kserve-controller/tests/integration/test_charm_ambient.py
+++ b/charms/kserve-controller/tests/integration/test_charm_ambient.py
@@ -19,7 +19,6 @@ from charmed_kubeflow_chisme.testing import (
     assert_metrics_endpoint,
     assert_security_context,
     deploy_and_assert_grafana_agent,
-    deploy_and_integrate_service_mesh_charms,
     get_alert_rules,
     get_pod_names,
 )
@@ -146,9 +145,9 @@ async def test_build_and_deploy(ops_test: OpsTest, request):
     )
 
     await ops_test.model.integrate(
-            f"{ISTIO_INGRESS_K8S_APP}:{ISTIO_INGRESS_ROUTE_ENDPOINT}",
-            f"{APP_NAME}:{ISTIO_INGRESS_ROUTE_ENDPOINT}",
-        )
+        f"{ISTIO_INGRESS_K8S_APP}:{ISTIO_INGRESS_ROUTE_ENDPOINT}",
+        f"{APP_NAME}:{ISTIO_INGRESS_ROUTE_ENDPOINT}",
+    )
     await ops_test.model.integrate(
         f"{ISTIO_BEACON_K8S_APP}:{SERVICE_MESH_ENDPOINT}",
         f"{APP_NAME}:{SERVICE_MESH_ENDPOINT}",

--- a/charms/kserve-controller/tests/integration/test_charm_ambient.py
+++ b/charms/kserve-controller/tests/integration/test_charm_ambient.py
@@ -58,10 +58,7 @@ from tests.integration.utils import (
 
 # ambient-mode Istio:
 ISTIO_K8S_APP = "istio-k8s"
-ISTIO_INGRESS_K8S_APP = "istio-ingress-k8s"
 ISTIO_BEACON_K8S_APP = "istio-beacon-k8s"
-ISTIO_INGRESS_ROUTE_ENDPOINT = "istio-ingress-route"
-ISTIO_INGRESS_GATEWAY_ENDPOINT = "gateway-metadata"
 SERVICE_MESH_ENDPOINT = "service-mesh"
 
 # tenacity
@@ -108,11 +105,6 @@ async def test_build_and_deploy(ops_test: OpsTest, request):
         trust=True,
     )
     await ops_test.model.deploy(
-        ISTIO_INGRESS_K8S_APP,
-        channel=istio_channel,
-        trust=True,
-    )
-    await ops_test.model.deploy(
         ISTIO_BEACON_K8S_APP,
         channel=istio_channel,
         trust=True,
@@ -144,10 +136,6 @@ async def test_build_and_deploy(ops_test: OpsTest, request):
         trust=True,
     )
 
-    await ops_test.model.integrate(
-        f"{ISTIO_INGRESS_K8S_APP}:{ISTIO_INGRESS_ROUTE_ENDPOINT}",
-        f"{APP_NAME}:{ISTIO_INGRESS_ROUTE_ENDPOINT}",
-    )
     await ops_test.model.integrate(
         f"{ISTIO_BEACON_K8S_APP}:{SERVICE_MESH_ENDPOINT}",
         f"{APP_NAME}:{SERVICE_MESH_ENDPOINT}",

